### DISCUSSION TITLE
Quarto rewrites raw LaTeX table to move caption to top when using crossrefs.

### DISCUSSION THREAD

**Discussion by aaronrudkin** [quarto-dev/quarto-cli#10278] [Q&A]
### Description

I am writing a document in quarto. In it, I include a LaTeX table.

```qmd
Hello world. In @tbl-main, I show blah blah blah

\begingroup
\centering
\begin{table}
   \begin{tabular}{lccccc}
      \tabularnewline \midrule \midrule
      Dependent Variable: & \multicolumn{5}{c}{test}\\
      Model:                       & (1)          & (2)          & (3)          & (4)          & (5)\\  
      \midrule
      \emph{Variables}\\
      Variable                       & 0.08$^{***}$ & 0.09$^{***}$ & 0.09$^{***}$ & 0.09$^{***}$ & 0.09$^{***}$\\
      \midrule
      Observations                 & 1,234  & 1,234  & 1,234   & 1,234   & 1,234 \\  
   \end{tabular}
   \caption{\label{tbl-main} This is a caption. Captions are great.}
\end{table}
\par\endgroup
```

When I compile to .pdf, the reference works and the table shows up, but the caption is moved from the bottom of the table to the top. I have tried overriding this by using the LaTeX caption package and forcing _all_ captions to below tables. No good. If I keep the .tex file, I notice that Quarto actually re-writes the end group, centering, table, tabular section as well.

If I rename the table's label to "table-main", this does not occur. In other words, it only occurs when the cross-ref system picks up on the table.

Is there a way to avoid this behavior?

Additionally, if the table is wrapped in a `\resizebox{}` (to, say, shrink width) this is also discarded by the rewrite process.

(In my real example the LaTeX is in a separate .tex file which is included using the include directive, but it doesn't seem to matter for the reprex)

**Comment by mcanouil** [comment]
You should use raw block to include raw code as much as possible.

Regarding the caption location, see <https://quarto.org/docs/authoring/tables.html#caption-location>.

There is currently no option to disable processing especially in the context of cross-reference system.
Quarto needs to modify the code to make it cross-refereanceable.

**Comment by aaronrudkin** [comment]
Thanks. The Caption argument fixes the caption problem at least, but it sounds like there is no way to stop Quarto from re-writing other parts of the table if I want to use cross-references, which is a bummer. Appreciate it!

**Reply by mcanouil** [reply]
> but it sounds like there is no way to stop Quarto from re-writing other parts of the table if I want to use cross-references,

Which parts exactly are modified in a way that you don't get what you want?

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/75b055f1-083e-49d8-ac38-d6d911e2d316">

It seems pretty alike to me.

**Reply by aaronrudkin** [reply]
> > but it sounds like there is no way to stop Quarto from re-writing other parts of the table if I want to use cross-references,
> 
> Which parts exactly are modified in a way that you don't get what you want?
>  ...
> It seems pretty alike to me.

I included this at the end of the issue but it was an edit a minute or two after posting: in practice, several of my tables are very wider tables are wrapped in `\resizebox{}` to shrink their width to page width. When Quarto rewrites the table to handle the Crossrefs, it removes that wrapper.

**Reply by mcanouil** [reply]
Could you share a **small self-contained "working" (reproducible)** example to work with, _i.e._, a complete Quarto document or a Git repository? The goal is to make it as easy as possible for us to recreate your problem so that we can fix it: please help us help you! Thanks.

---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

{{< lipsum 1 >}}

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image]({{< placeholder 600 400 >}}){#fig-placeholder}

{{< lipsum 1 >}}

The end after @fig-placeholder.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.
